hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/endpoint/TestOverrideResponseHeader.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.hadoop.ozone.client.OzoneClient;
+import org.apache.hadoop.ozone.client.OzoneClientStub;
+import org.apache.hadoop.ozone.client.io.OzoneOutputStream;
+import org.apache.hadoop.ozone.s3.exception.OS3Exception;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.mockito.Mockito.doReturn;
+
+public class TestOverrideResponseHeader {
+  public static final String CONTENT = "0123456789";
+  public static final String ContentType1 = "video/mp4";
+  public static final String ContentType2 = "text/html; charset=UTF-8";
+  public static final String ContentLanguage1 = "en-CA";
+  public static final String ContentLanguage2 = "de-DE, en-CA";
+  public static final String Expires1 = "Wed, 21 Oct 2015 07:29:00 GMT";
+  public static final String Expires2 = "Wed, 21 Oct 2015 07:28:00 GMT";
+  public static final String CacheControl1 = "no-cache";
+  public static final String CacheControl2 = "max-age=604800";
+  public static final String ContentDisposition1 = "inline";
+  public static final String ContentDisposition2 = "attachment; "
+      + "filename=\"filename.jpg\"";
+  public static final String ContentEncoding1 = "gzip";
+  public static final String ContentEncoding2 = "compress";
+  HttpHeaders headers;
+  ContainerRequestContext context;
+
+  ObjectEndpoint rest;
+
+  @Before
+  public void init() throws IOException {
+    OzoneClient client = new OzoneClientStub();
+    client.getObjectStore().createS3Bucket("b1");
+    OzoneBucket bucket = client.getObjectStore().getS3Bucket("b1");
+    OzoneOutputStream keyStream =
+        bucket.createKey("key1", CONTENT.getBytes(UTF_8).length);
+    keyStream.write(CONTENT.getBytes(UTF_8));
+    keyStream.close();
+
+    rest = new ObjectEndpoint();
+    rest.setClient(client);
+
+    headers = Mockito.mock(HttpHeaders.class);
+    rest.setHeaders(headers);
+
+    context = Mockito.mock(ContainerRequestContext.class);
+    UriInfo uriInfo = Mockito.mock(UriInfo.class);
+    Mockito.when(context.getUriInfo()).thenReturn(uriInfo);
+    MultivaluedHashMap<String, String> queryParameter = new MultivaluedHashMap<>();
+    Mockito.when(context.getUriInfo().getQueryParameters())
+        .thenReturn(queryParameter);
+    rest.setContext(context);
+  }
+
+  private void setHeaders() {
+    doReturn(ContentType1).when(headers).getHeaderString("Content-Type");
+    doReturn(ContentLanguage1).when(headers).getHeaderString("Content-Language");
+    doReturn(Expires1).when(headers).getHeaderString("Expires");
+    doReturn(CacheControl1).when(headers).getHeaderString("Cache-Control");
+    doReturn(ContentDisposition1).when(headers).getHeaderString("Content-Disposition");
+    doReturn(ContentEncoding1).when(headers).getHeaderString("Content-Encoding");
+  }
+
+  @Test
+  public void testInheritRequestHeader() throws IOException, OS3Exception {
```

Comment:
Can you please add these new test cases in `TestObjectGet` instead?

Reply:
done

Reply:
Sorry, I meant both tests.